File: backend/src/main/java/com/urbanman/ecommerce/model/Coupon.java
package com.urbanman.ecommerce.model;

import jakarta.persistence.*;

@Entity
@Table(name = "coupons")
public class Coupon {

    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long couponId;

    @Column(unique = true, nullable = false)
    private String code;

    private Double discountPercent;   // e.g. 10.0 = 10%
    private Double discountAmount;    // flat discount in rupees
    private Double minOrderAmount;    // minimum order to apply
    private Boolean active = true;

    // Getters & Setters
    public Long getCouponId()                           { return couponId; }
    public void setCouponId(Long id)                    { this.couponId = id; }

    public String getCode()                             { return code; }
    public void setCode(String code)                    { this.code = code; }

    public Double getDiscountPercent()                  { return discountPercent; }
    public void setDiscountPercent(Double d)            { this.discountPercent = d; }

    public Double getDiscountAmount()                   { return discountAmount; }
    public void setDiscountAmount(Double d)             { this.discountAmount = d; }

    public Double getMinOrderAmount()                   { return minOrderAmount; }
    public void setMinOrderAmount(Double m)             { this.minOrderAmount = m; }

    public Boolean getActive()                          { return active; }
    public void setActive(Boolean active)               { this.active = active; }
}

File: backend/models/Coupon.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/db');

const Coupon = sequelize.define('Coupon', {
  couponId: {
    type: DataTypes.BIGINT,
    primaryKey: true,
    autoIncrement: true,
    field: 'coupon_id'
  },
  code: {
    type: DataTypes.STRING,
    allowNull: false,
    unique: true
  },
  discountPercent: {
    type: DataTypes.DOUBLE,
    allowNull: true,
    field: 'discount_percent'
  },
  discountAmount: {
    type: DataTypes.DOUBLE,
    allowNull: true,
    field: 'discount_amount'
  },
  minOrderAmount: {
    type: DataTypes.DOUBLE,
    allowNull: true,
    field: 'min_order_amount'
  },
  active: {
    type: DataTypes.BOOLEAN,
    allowNull: false,
    defaultValue: true
  }
}, {
  tableName: 'coupons',
  timestamps: false
});

module.exports = Coupon;
